### PR TITLE
feat(pose_estimator_arbiter): componentize PoseEstimatorArbiter

### PR DESCRIPTION
**Pull request by a-maumau**
## Description

Componentize the node in localization/pose_estimator_arbiter.
(Add glog feature to localization/pose_estimator_arbiter)

Also, log output is changed from "screen" to "both".

## Tests performed

Confirmed that terminating the `pose_estimator_arbiter` process will output logs to both the terminal and the log file.

```Text
1717132546.7531216 [pose_estimator_arbiter_node-79] [INFO] [1717132546.752725795] [rclcpp]: signal_handler(signum=15)
1717132546.7539675 [pose_estimator_arbiter_node-79] *** Aborted at 1717132546 (unix time) try "date -d @1717132546" if you are using GNU date ***
1717132546.7583516 [pose_estimator_arbiter_node-79] PC: @                0x0 (unknown)
1717132546.7589211 [pose_estimator_arbiter_node-79] *** SIGTERM (@0x3e8000e1b20) received by PID 2979663 (TID 0x7fbc82380680) from PID 924448; stack trace: ***
1717132546.7594931 [pose_estimator_arbiter_node-79]     @     0x7fbc82c1d4d6 google::(anonymous namespace)::FailureSignalHandler()
1717132546.7604535 [pose_estimator_arbiter_node-79]     @     0x7fbc82a19ded rclcpp::SignalHandler::signal_handler()
1717132546.7610266 [pose_estimator_arbiter_node-79]     @     0x7fbc82042520 (unknown)
1717132546.7615888 [pose_estimator_arbiter_node-79]     @     0x7fbc820912c0 (unknown)
1717132546.7616441 [pose_estimator_arbiter_node-79]     @     0x7fbc82098002 pthread_mutex_lock
1717132546.7622015 [pose_estimator_arbiter_node-79]     @     0x7fbc82965374 rclcpp::executors::MultiThreadedExecutor::run()
1717132546.7626779 [pose_estimator_arbiter_node-79]     @     0x7fbc82965808 rclcpp::executors::MultiThreadedExecutor::spin()
1717132546.7627404 [pose_estimator_arbiter_node-79]     @     0x557a1119a496 main
1717132546.7627900 [pose_estimator_arbiter_node-79]     @     0x7fbc82029d90 (unknown)
1717132546.7633214 [pose_estimator_arbiter_node-79]     @     0x7fbc82029e40 __libc_start_main
1717132546.7633750 [pose_estimator_arbiter_node-79]     @     0x557a1119ae45 _start
1717132546.7710996 [ERROR] [pose_estimator_arbiter_node-79]: process has died ...
```

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [ ] I've confirmed the [contribution guidelines].
- [ ] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by a-maumau** [comment]
@KYabuuchi 
I am doing the componentizing on the `pose_estimator_arbiter`.

It seems running the following command from https://github.com/autowarefoundation/autoware.universe/pull/6144 
```sh
ros2 launch autoware_launch logging_simulator.launch.xml \
    map_path:=$HOME/MAP vehicle_model:=sample_vehicle sensor_model:=awsim_sensor_kit \
    pose_source:=ndt_yabloc_eagleye_artag # or ndt_yabloc, eagleye_artag and so on
```

, the `pose_estimator_arbiter_node` process is running, but the ROS node `pose_estimator_arbiter` itself is not running.

In the launch log,
```
1717122685.1253524 [INFO] [pose_estimator_arbiter_node-79]: process started with pid [2848605]
...
1717122700.6052625 [WARNING] [launch]: user interrupted with ctrl-c (SIGINT)
...
1717122703.2054548 [INFO] [pose_estimator_arbiter_node-79]: process has finished cleanly [pid 2848605]
```

Also, running the test: `colcon test --event-handlers console_cohesion+ summary+ --packages-select pose_estimator_arbiter` fails on the `/test_pose_estimator_arbiter.py` with
```
.., line 170, in test_node_link
self.assertGreater(len(self.ndt_relayed), 5)
AssertionError: 0 not greater than 5
```
This also should indicating the node is not running.

Do you have any suggestion or solution for this?

**Comment by KYabuuchi** [comment]
@a-maumau I will check soon. Please wait :pray:

File: localization/pose_estimator_arbiter/CMakeLists.txt [hunk 1]
```diff
@@ -4,29 +4,23 @@ project(pose_estimator_arbiter)
...
   src/pose_estimator_arbiter/pose_estimator_arbiter_core.cpp
-  src/pose_estimator_arbiter/pose_estimator_arbiter_node.cpp
+  src/pose_estimator_arbiter/switch_rule/enable_all_rule.cpp
 )
 target_include_directories(${PROJECT_NAME} PUBLIC src)
-target_link_libraries(${PROJECT_NAME} switch_rule glog::glog)
 
-# ==============================
+rclcpp_components_register_node(${PROJECT_NAME}
+  PLUGIN "PoseEstimatorArbiter"
```
Comment (KYabuuchi) [inline localization/pose_estimator_arbiter/CMakeLists.txt:19]: @a-maumau Please try to replace
```suggestion
  PLUGIN "pose_estimator_arbiter::PoseEstimatorArbiter"
```

Comment (KYabuuchi) [reply localization/pose_estimator_arbiter/CMakeLists.txt:19]: In my environment, the above suggestion has fixed all issues which you encountered.

Comment (a-maumau) [reply localization/pose_estimator_arbiter/CMakeLists.txt:19]: Thank you very mush for checking out the codes.
I also confirmed the above suggestion works.

**Review by KYabuuchi — APPROVED** [review]
Thank you for your contributions! LGTM

**Pull request merged**